python/ray/data/_internal/execution/operators/map_transformer.py

```diff
@@ -47,7 +46,10 @@ def __init__(
 
     @abstractmethod
     def __call__(
-        self, input: Iterable[MapTransformFnData], ctx: TaskContext
+        self,
+        input: Iterable[MapTransformFnData],
+        target_max_block_size: int,
```

Comment:
nit, no need to change this function signature. You can just add this to the init parameters of `BuildOutputBlocksMapTransformFn`

Reply:
Hmm I did this because the map transformers are created before the optimizer rules are applied, so the target block size may change later.

Reply:
We can save the target size as an attribute in BuildOutputBlocksMapTransformFn and then change it with an optimizer rule.